src/board-explorer-app/board-explorer-app.html

```diff
@@ -1179,15 +1183,30 @@
 
     onFiltersChanged: function(board, filters) {
       var href = window.location.href.replace(/#.*/, ""); /* Trim of # */
-
       /* If the board is set, add it to the URL */
+
+      this.$.boardViewer.removeFilter("category");
+
       if (board) {
          href += "#" + board;
       }
 
+      if (this.category) {
+         href += "/" + this.category.slug;
+      }
+
       /* If the filters are set, add them to the URL */
       if (board && filters && filters.length) {
-        href += "/" + this.filters.join();
+        for (var i = 0; i < this.categories.length; i++) {
+          var index = filters.indexOf(this.categories[i].slug);
+          if (index != -1) {
+            filters.splice(index, 1);
+          }
+        }
+
```

Comment:
Am I following the `for (var i = 0; i < this.categories.length; i++) {...}` section of code correctly in that it is removing all category names from the filters array (since the category is added to the board-viewer via it's own addFilter call?)